docs/src/pages/components/date-picker/date-picker.md

```diff
@@ -0,0 +1,105 @@
+import LocalizationProvider from '@material-ui/lab/LocalizationProvider';
+
+function App() {
+  return (
+    <LocalizationProvider dateAdapter={DateFnsAdapter}>
+      ...
+    </LocalizationProvider>
+  );
+}
+```
+
+## Basic usage
+
+The date picker will be rendered to modal dialog on mobile and textfield with popover on desktop.
+
+{{"demo": "pages/components/date-picker/BasicDatePicker.js"}}
+
+## Responsiveness
+
+The date picker component is designed and optimized for the device it runs on.
+
+- The "Mobile" version works best for touch devices and small screens.
+- The "Desktop" version works best for mouse devices and large screens.
+
+By default, the `DatePicker` component uses a `@media (pointer: fine)` media query to determine which version to use.
+This can be customized with the `desktopModeMediaQuery` prop.
+
+{{"demo": "pages/components/date-picker/ResponsiveDatePickers.js"}}
+
+## Localization
+
+Use `LocalizationProvider` to change the date-engine locale that is used to render the date picker. Here is an example of changing the locale for the `date-fns` adapter:
+
+{{"demo": "pages/components/date-picker/LocalizedDatePicker.js"}}
+
+## Views playground
+
+It is possible to combine `year`, `month`, and `date` selection views. Views will appear in the order they passed to the `views` array.
+
+{{"demo": "pages/components/date-picker/ViewsDatePicker.js"}}
+
+## Static mode
+
+It is possible to render any picker without the modal/popover and text field. This can be helpful to build custom popover/modal containers.
+
+{{"demo": "pages/components/date-picker/StaticDatePickerDemo.js", "bg": true}}
+
+## Landscape orientation
+
+For ease of use the date picker will automatically change the layout between portrait and landscape by subscription to the `window.orientation` change. You can force a specific layout using the `orientation` prop.
+
+{{"demo": "pages/components/date-picker/StaticDatePickerLandscape.js", "bg": true}}
+
+## Sub-components
+
+Some lower level sub-components (`DayPicker`, `MonthPicker` and `YearPicker`) are also exported. These are rendering without a wrapper or outer logic (masked input, date values parsing and validation, etc.).
+
+{{"demo": "pages/components/date-picker/InternalPickers.js"}}
+
+## Custom input component
+
+You can customize rendering of the input with the `renderInput` prop. Make sure to spread `ref` and `inputProps` correctly to the custom input component.
+
+{{"demo": "pages/components/date-picker/CustomInput.js"}}
+
+## Customized day rendering
+
+The displayed days are customizable with the `renderDay` function prop.
+You can take advantage of the internal [PickersDay](/api/pickers-day) component.
+
+{{"demo": "pages/components/date-picker/CustomDay.js"}}
+
+## Dynamic data
+
+Sometimes it may be necessary to display additional info right in the calendar. Here is an example of prefetching and displaying server-side data using the `onMonthChange`, `loading`, and `renderDay` props.
```

Comment:
```suggestion
Sometimes it may be necessary to display additional info right in the calendar. Here's an example of prefetching and displaying server-side data using the `onMonthChange`, `loading`, and `renderDay` props.
```